pyfar/classes/audio.py

```diff
@@ -1296,64 +1300,63 @@ def _arithmetic(data: tuple, domain: str, operation: Callable, **kwargs):
     return result
 
 
-def _assert_match_for_arithmetic(data: tuple, domain: str, division: bool,
-                                 matmul: bool):
-    """Check if type and meta data of input is fine for arithmetic operations.
-
-    Check if sampling rate and number of samples agree if multiple signals are
-    provided. Check if arrays are numeric. Check if a power signal is contained
-    in the input. Extract cshape of result.
+def _assert_match_for_arithmetic(data, domain, division, matmul):
+    """
+    Validates that the type and metadata of input data
+        are appropriate for performing arithmetic operations.
 
-    Input:
+    Parameters
+    ----------
     data : tuple
-        Can contain Signal, TimeData, FrequencyData, and array like data
+        A tuple containing data elements, which can be instances of
+        Signal, TimeData, FrequencyData, or array-like objects.
     domain : str
-        Domain in which the arithmetic operation should be performed. 'time' or
-        'freq'.
+        The domain in which the arithmetic operation is to be performed.
+        Either 'time' or 'freq'.
     division : bool
-        ``True`` if a division is performed, ``False`` otherwise
-    matmul: bool
-        ``True`` if a  matrix multiplication is performed, ``False`` otherwise
+        Indicates if the operation is a division.
+    matmul : bool
+        Indicates if the operation is a matrix multiplication.
 
     Returns
     -------
-    sampling_rate : number, None
-        Sampling rate of the signals. None, if no signal is contained in `data`
-    n_samples : number, None
-        Number of samples of the signals. None, if no signal is contained in
-        `data`
-    fft_norm : str, None
-        FFT norm of the first signal in `data`, if all FFT norms are None.
-        Otherwise the first FFT norm that is not None is taken.
-    times : numpy array, None
-        The times if a TimeData object was passed. None otherwise.
-    frequencies : numpy array, None
-        The frequencies if a FrequencyData object was passed. None otherwise.
-    audio_type : type, None
-        Type of the audio class if contained in data. Otherwise None.
-    cshape : tuple, None
-        Largest channel shape of the audio classes if contained in data.
-        Otherwise empty tuple.
```

Comment:
I prefer the explicit documentation with a description of each return parameter.